pytorch_lightning/utilities/distributed.py

```diff
@@ -239,10 +244,10 @@ def all_gather_ddp_if_available(
 
 
 def register_ddp_comm_hook(
-    model: DistributedDataParallel,
+    model: Type[DistributedDataParallel],
```

Comment:
I don't understand this change. The code shows we are passing in an instance so I would expect the previous type hint was correct.

Reply:
In the previous case, mypy throws the following error
```
pytorch_lightning/utilities/distributed.py:343: error: "Tensor" not callable  [operator]
```
when `model.register_comm_hook(state=ddp_comm_state, hook=ddp_comm_hook)` being called.

However, I agree with you that the previous type hint looks correct. Maybe should we ignore mypy here?

Reply:
There must be a bug somewhere in the typing where `model.register_comm_hook` becomes a tensor :( 
But in any case with the new type annotation, we get: 
`pytorch_lightning/utilities/distributed.py:343: error: "Type[DistributedDataParallel]" has no attribute "register_comm_hook"`

so we should not do it and just leave as is imo

Reply:
@SciPioneer have you seen this typing issue come up elsewhere for DDP comm hooks?

Reply:
Hmmm... I don't know why mypy throws an error on my local while it's okay here (gonna check my versions). But bottom line is, mypy typing should be fine for now :)